std/zip.d

```diff
@@ -387,6 +389,86 @@ final class ArchiveMember
         printf("\tindex = x%08x\n", index);
     }
     }
+
+private:
+    void enforceIntegrity()
+    {
+        import std.string : lastIndexOf;
+        import std.algorithm.iteration : splitter;
+        import std.algorithm.searching : all;
+        import std.ascii : isAlpha;
+
+        // Most filesystems do not allow for more than 255 characters in a file name.
+        // Longer names are probably malware and we reject them therefore.
+        enforce!ZipException(name.length - (name.lastIndexOf('/') + 1) <= 255,
+                             "contains filename with length > 255");
+
+        // ASCII characters < 0x20 and 0x7F can be used to hide path traversal attacks.
+        // Therefore we reject them. Backslash is not allowed according to the specs.
+        enforce!ZipException(name.all!(c => c >= 0x20 && c != 0x7F && c != '\\'),
+                             "filename contains invalid symbol");
```

Comment:
UNIX filenames can contain ANY characters except NUL and /, so not sure about this.

Reply:
I'm not sure on this either. The source (see above) writes this:

> // CVE-2003-0282 (aka "JELMER")
> // Some zip implementations filter control characters at the least.
> // This behavior can be exploited to mask ".." in a directory traversal.
> // https://www.securityfocus.com/archive/1/321090

Reply:
Hmm, that looks like a bug in the unzip utility (it filters out the control characters) or the operating system (canonicalizes them out and interprets `..` after this canonicalization). If it's the former, then we don't need to do anything. If it's the latter, then this solution is inadequate - we need to run the path through the same algorithm the OS uses for canonicalization and check the result.

Reply:
ditto? By the way, ISO rejects some of these characters. I'll address this in a separate PR, cause here it's only about validating not rejecting.